File: Leetcode/Easy/1. Two Sum/solution.php
class Solution {

/**
 * @param Integer[] $nums
 * @param Integer $target
 * @return Integer[]
 */
 function twoSum(array $nums, int $target) : array {
    $map = [];

    foreach($nums as $index => $num) {
        $complement = $target - $num;
        if(isset($map[$complement]))
            return [$map[$complement], $index];
        $map[$num] = $index;
    }
    return [];
 }
}

File: Leetcode/Easy/1. Two Sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashMap = {}
        for index, num in enumerate(nums):
            complement = target - num
            if complement in hashMap:
                return [hashMap[complement], index]
            hashMap[num] = index
            
